src/agent/secrets.clj

```diff
@@ -74,6 +75,41 @@
           [nil (format "Secret '%s' not found" secret-path)]))))
 
 
+;; filesystem
+(declare get-filesystem-secret)
+
+(def default-filesystem-location
+  "secrets.json")
+
+(defn persist-secret [secret]
+  (log/info "Persisting filesystem secrets")
+  (try
+    (spit default-filesystem-location secret)
+    [secret nil]
+    (catch Exception e
+      (log/error (format "Failed to persist filesystem secrets with error: %s" e))
+      [nil "failed to persist filesystem secrets"])))
+
+(defmethod fetch "filesystem" [task]
+  (log/info (format "Fetching secrets from filesystem for task id: %s" (:id task)))
+  (err/err->> task
+              validate-secret-path
+              get-filesystem-secret
+              (fn[result] [(assoc result :secrets ((keyword (:secret-path task)) (:secrets result)))])))
+
+(defn get-filesystem-secret [task]
+  (try
+    (if-let [content (slurp default-filesystem-location)]
+      (err/err->> content
+                  json->map
+                  (fn [secrets] [(assoc task :secrets secrets) nil]))
+      (do (log/error (format "failed to get filesystem '%s'" (:secret-path task)))
```

Comment:
You could raise an exception instead with the message: "empty filesystem content". It will be caught in the block below

Reply:
Not sure if I got your point. 

This method follows the same pattern of all other secret-provider. It tries to do something, and if it fails, returns an error using the `[nil "error"]` notation. The error will be handled by the caller, that will stop the execution of the task and return the error msg to the user.

Let me know if I misunderstood the suggestion.

Reply:
This is redundant because you're already returning the same error when an exception is found. Thus raising an exception will default to the exception block. 
But isn't a problem.